### ISSUE TITLE
Implement memory protection APIs (RW/RX/RO)

### ISSUE THREAD

**Issue by roblabla** [reswitched/libtransistor#119]
Add a JIT API to libtransistor. It'd be really awesome to implement the `mprotect` API, such that emulators using that syscall can work as-is without modifications.

**Comment by iOS4all** [comment]
This is really.
Please can we get vulkan api to work besides GPU rendering?
Thanks a lot for your hard work

**Comment by allanxp4** [comment]
JIT as in a Just In Time compiler? 
Do you mean an API like libgccgit does?

**Comment by roblabla** [comment]
By JIT, we really mean memory protection APIs. AKA the ability to make a memory region go from RW to RX and back.

**Comment by allanxp4** [comment]
May I suggest renaming the issue? *Very* few people know that as "JIT" (and I still don't know what that means), but as memory protection, DEP, NX, etc. Even searching for JIT on osdev doesn't return anything related to memory protection. (not bashing your work, just trying to get more people to contribute :) )

By the way, i'm getting familiar with the library and the inner workings of the horizon, looks like, a really nice microkernel. done a few tests on the switch, probably gonna try implemting memory protection later once I really get how things work. Maybe documeting a few things in the way.

**Comment by mailwl** [comment]
it has the same mean on iPhone - Safari, or even WebKit uses it for JavaScript JIT, i think, HOS uses it for JS too (in WebKit) - impossible to sign code remotely, which auto generated on the fly

**Comment by allanxp4** [comment]
Yes, but you use it for JIT, it isn't a JIT API, it is a memory protection API. You use it to make pages executable in order to make the code generated by your JIT engine executable, but it isn't a JIT in itself.
Most people will come here thinking in implementing a libgccjit or a DynASM, not an mprotect.

**Comment by mailwl** [comment]
JIT memory - RWX+disable codesign check. it's all

**Comment by roblabla** [comment]
@allanxp4 you're absolutely right. Mostly called it JIT API because that's what libnx called it, but it's a pretty misleading name :).

**Comment by leo60228** [comment]
Need this for something I'm working on. Not sure how to work around this.
